backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/i18n/po/POTrlRepository.java

```diff
@@ -255,11 +287,71 @@ else if (po.isDynAttributeTrue(DYNATTR_TrlUpdateMode_UpdateIdenticalTrls))
 
 		final int updatedCount = DB.executeUpdateEx(sql.toString(), ITrx.TRXNAME_ThreadInherited);
 		logger.debug("Updated {} translation records for {}", updatedCount, po);
+		final CacheMgt cacheMgt = CacheMgt.get();
+		cacheMgt.reset(toTrlTableName(tableName), po.get_ID());
 
-		//
 		return updatedCount >= 0;
 	}
 
+	public void updateBaseTableOnTranslationChange(@NonNull final PO po)
+	{
+		final POTrlInfo trlInfo = po.getPOInfo().getTrlInfo();
+		if (!trlInfo.isTranslations())
+		{
+			return;
+		}
+
+		final String trlTableName = po.get_TableName();
+		final String baseTableName = toBaseTableName(trlTableName);
+		if(DB.isDBColumnPresent(baseTableName, COLUMNNAME_AD_Element_ID))
+		{
+			return;
+		}
+
+		final String keyColumnName;
+		final String keyColumnValue;
+		final String languageColumnValue;
+		if(trlInfo.getKeyColumnNames().get(0).equalsIgnoreCase(I_AD_Language.COLUMNNAME_AD_Language))
+		{
```

Comment:
don't get this "if" and why we have multiple key columns

Reply:
it's for _trl tables which have 2 keys and this check is for the case if the order of ad_language and base_table primary key isn't always the same.

Reply:
the purpose of POTrlInfo is to describe a base table, tell which is the trl table, what columns are translated etc,

and not to have a POTrlInfo for each table, no matter if is a base table or a trl table